c7n/resources/redshift.py

```diff
@@ -336,14 +336,30 @@ def process(self, clusters):
                         "Exception creating Redshift snapshot  \n %s",
                         f.exception())
         return clusters
-
     def process_cluster_snapshot(self, cluster):
         c = local_session(self.manager.session_factory).client('redshift')
+        for r in cluster:
+            arn = self.manager.generate_arn(cluster['ClusterIdentifier'])
+            cluser_tags_list = c.describe_tags(ResourceName=arn)
```

Comment:
there's a few issues here.. first, `for r in cluster`, is basically key iteration on a cluster instance attributes, which makes no sense. second, the tags are already present on the redshift cluster resource, ie. custodian already fetched them, so this refetch is redundant.